### PR TITLE
creates base version files. tags & pushes base repo

### PR DESCRIPTION
**Pull request by Bharath92**
https://github.com/Shippable/beta/issues/538

### CODE REVIEW

File: tagAndPushBaseRepo.sh [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  git add $migrations_version_file
+  git add $migrations_post_install_versions_file
+  git add $migrations_pre_install_versions_file
+  git add $versions_file
+
+  git commit -m "creating $RES_VER_NAME version files"
+
+  git push up master
+
+  IMG_REPO_COMMIT_SHA=$(git rev-parse HEAD)
```
Comment (Bharath92) [inline tagAndPushBaseRepo.sh]: This has been added here so that, while tagging, the checkout happens to the correct commit.

**Comment by ric03uec** [comment]
@Bharath92 
- we're not using `git rev-parse HEAD` in the final `tag-push-*` jobs. this is because the HEAD for repo might not be the one that is deployed to RC and can lead to incorrect tags on the repo
- the pipeline is set up in following way
    - for any repo, push the image where the image tag is sha tag
    - deploy that image on rc
    - when creating the release, use the same sha to tag the repo with release version
- you should look at the pipeline  for `cexec` as a reference for this. even though `cexec` image is never deployed anywhere, we're creating cexec pipeline the exact same way as other components for sake of consistency

**Comment by Bharath92** [comment]
closing this PR after discussion with @ric03uec

**Comment by avinci** [comment]
This also does not take into account rerunning of the job. That file might already exist.

**Pull request closed without merging**